### PR TITLE
feat: add wad_ray_math with tests

### PR DESCRIPTION
**Pull request by FabienCoutant**
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Handle Wad & Ray mul/div 

Issue Number: #211 

## What is the new behavior?

Add new functions to handle 18 & 27 decimals base maths

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: src/math/src/tests/wad_ray_math_test.cairo [hunk 1]
```diff
@@ -0,0 +1,160 @@
+use core::debug::PrintTrait;
```
Comment (gaetbout) [inline src/math/src/tests/wad_ray_math_test.cairo:1]:
```suggestion
```

File: src/math/src/wad_ray_math.cairo [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    return (a * RAY + half_b) / b;
+}
+
+/// Casts ray down to wad
+/// # Arguments
+/// * a Ray
+/// # Returns
+/// * a converted to wad, rounded half up to the nearest wad
+fn ray_to_wad(a: u256) -> u256 {
+    let half_ratio = WAD_RAY_RATIO / 2;
```
Comment (gaetbout) [inline src/math/src/wad_ray_math.cairo:91]: might wanna hardcode that to avoid unnecessary computation

**Review by gaetbout — APPROVED** [review]
otherwise, LGTM 👍

File: src/math/src/tests/wad_ray_math_test.cairo (hunk 1 above)
Comment (FabienCoutant) [reply src/math/src/tests/wad_ray_math_test.cairo]: I thought that scarb format will run on pre-commit. It's removed

File: src/math/src/wad_ray_math.cairo (hunk 2 above)
Comment (FabienCoutant) [reply src/math/src/wad_ray_math.cairo]: It saving fee gas, agree then

File: src/math/src/tests/wad_ray_math_test.cairo (hunk 1 above)
Comment (LucasLvy) [reply src/math/src/tests/wad_ray_math_test.cairo]: scarb format only formats it doesn't remove unused imports. Also i don't think we have any pre commit hook on this repo do we ?

**Review by LucasLvy — APPROVED** [review]

File: src/math/src/tests/wad_ray_math_test.cairo (hunk 1 above)
Comment (FabienCoutant) [reply src/math/src/tests/wad_ray_math_test.cairo]: Running `scarb format` locally had fix import order at least. Didn't really check if useless import has been removed

**Pull request merged**